### PR TITLE
feat: add --auto mode to TUI for bypassing permission prompts for single session

### PR DESCRIPTION
**Pull request by evanjacobson**
## Summary

- Adds auto mode to the interactive TUI, allowing all permission requests to be auto-approved without prompting
- Activatable via `kilo --auto` CLI flag, `/auto` slash command, or `ctrl+shift+a` keybind
- Persistent warning banner below the chat textbox when active
- Non-persisted — resets on TUI restart for safety

## How it works

**Dual-layer auto-approval architecture:**

1. **SyncProvider intercept** — When a `permission.asked` SSE event arrives while auto mode is on, it immediately replies `"once"` and skips adding the permission to the reactive store. This eliminates the race condition where permissions arrive before the Session component mounts.

2. **Session effect fallback** — A `createEffect` in the Session component drains any permissions already in the store when auto mode is toggled on mid-session. Uses a `Set<string>` to prevent duplicate replies.

Both layers use `reply: "once"` (not `"always"`) so no persistent allow rules are created that survive toggling auto mode off.

## Activation methods

| Method | Usage |
|---|---|
| CLI flag | `kilo --auto` |
| Slash command | `/auto` in the prompt |
| Keybind | `ctrl+shift+a` (configurable via `keybinds.auto_toggle`) |
| Command palette | `ctrl+p` → "Enable auto mode" |

## UI

- Warning-colored banner below the chat textbox: `△ Auto mode — permissions are auto-approved  (ctrl+shift+a to toggle)`
- Startup toast when `--auto` flag is used
- Prompt remains visible and enabled during auto-approval (no flicker)
image
image
image
image

## Files changed

| File | Change |
|---|---|
| `src/cli/cmd/tui/context/auto.tsx` | **New** — `AutoModeProvider` context with non-persisted signal |
| `src/cli/cmd/tui/thread.ts` | `--auto` CLI flag |
| `src/cli/cmd/tui/context/args.tsx` | `auto` field on `Args` interface |
| `src/cli/cmd/tui/app.tsx` | Provider tree, `/auto` command, startup toast |
| `src/cli/cmd/tui/context/sync.tsx` | Layer 1 — SSE event intercept |
| `src/cli/cmd/tui/routes/session/index.tsx` | Layer 2 — Session effect + Prompt visible/disabled fix |
| `src/cli/cmd/tui/component/prompt/index.tsx` | Persistent banner |
| `src/config/config.ts` | `auto_toggle` keybind (default `ctrl+shift+a`) |
| `test/kilocode/auto-mode.test.ts` | **New** — 14 tests covering context logic, permission reply semantics, deduplication, config schema, and edge cases |

## Test plan

- 14 tests, all passing (`bun test test/kilocode/auto-mode.test.ts`)
- Typecheck clean (`bun run typecheck`)
- Manually verified in TUI: flag, slash command, keybind, banner, toast

### CODE REVIEW

**Comment by marius-kilocode** [comment]
Feedback from our product history:

- `--auto` mode would go into the direction of cloud agents (which is not intended to be used by end users only for CI)
- `--yolo` mode used to be the mode you are implementing here. So most of our previous users know that terminology instead

Further questions:
- How did you handle questions in this PR?
- Ideally we give the user the choice to configure the sensitivity of how much the CLI asks (but maybe out of scope for this PR)

File: packages/opencode/src/cli/cmd/tui/context/sync.tsx [hunk 1]
```diff
@@ -127,6 +129,33 @@ export const { use: useSync, provider: SyncProvider } = createSimpleContext({
 
         case "permission.asked": {
           const request = event.properties
+          // kilocode_change start - auto-approve immediately, fall back to store on failure
+          if (auto.enabled()) {
```
Comment (alex-alecu) [inline packages/opencode/src/cli/cmd/tui/context/sync.tsx:133]: `KILO_PERMISSION='{"*":"allow"}'` and config `"permission": "allow"` already auto-approve everything at the server level — no `permission.asked` events are even emitted. Could `--auto` set the session permission rules to `allow` server-side instead of intercepting each event client-side?

File: packages/opencode/src/cli/cmd/tui/context/sync.tsx [hunk 2]
```diff
@@ -127,6 +129,33 @@ export const { use: useSync, provider: SyncProvider } = createSimpleContext({
...
+            sdk.client.permission
+              .reply({
+                reply: "once",
```
Comment (alex-alecu) [inline packages/opencode/src/cli/cmd/tui/context/sync.tsx:136]: `kilo run --auto` replies `"always"`, which creates a rule so matching patterns are auto-approved going forward. Using `"once"` here means every single tool call needs its own round-trip. Why the difference?

File: packages/opencode/src/cli/cmd/tui/routes/session/index.tsx [hunk 3]
```diff
@@ -202,6 +205,36 @@ export function Session() {
   const toast = useToast()
   const sdk = useSDK()
 
+  // kilocode_change start - auto-approve permissions when auto mode is active
+  const replied = new Set<string>()
```
Comment (alex-alecu) [inline packages/opencode/src/cli/cmd/tui/routes/session/index.tsx:209]: `replied` and `inflight` grow forever and are never cleared when auto mode is toggled off. Could this leak memory over a long session?

File: packages/opencode/src/cli/cmd/tui/routes/session/index.tsx [hunk 4]
```diff
@@ -202,6 +205,36 @@ export function Session() {
...
+        .reply({
+          reply: "once",
+          requestID: request.id,
+        })
+        .then(() => {
+          replied.add(request.id)
+        })
+        .catch((err) => {
+          console.error("auto-reply failed, retrying in 2s", err)
+          setTimeout(() => setRetryTick((n) => n + 1), 2000)
```
Comment (alex-alecu) [inline packages/opencode/src/cli/cmd/tui/routes/session/index.tsx:229]: If the server stays unreachable, doesn't this retry forever with no backoff and no max attempts?

File: packages/opencode/test/kilocode/auto-mode.test.ts [hunk 5]
```diff
@@ -0,0 +1,414 @@
...
+
+// ── AutoMode context logic ──────────────────────────────────────────────────
+// Tests the core signal/toggle behavior that AutoModeProvider uses.
+// We replicate the context's init logic directly since SolidJS SSR build
+// (used in bun test) evaluates signals synchronously.
+
+describe("auto mode context", () => {
+  test("defaults to disabled", () => {
+    createRoot((dispose) => {
+      const [enabled] = createSignal(false)
```
Comment (alex-alecu) [inline packages/opencode/test/kilocode/auto-mode.test.ts:22]: This creates a local `createSignal(false)` and checks it returns `false`. Isn't it testing SolidJS itself rather than `AutoModeProvider`? Same for the two tests below.

**Comment by evanjacobson** [comment]
Closing for now, but may pick it back up when everyone has more bandwidth
